### DISCUSSION TITLE
addons inside this plugin

### DISCUSSION THREAD

**Discussion by LeonelRFF** [st3v3nmw/obsidian-spaced-repetition#1050] [Q&A] [answered]
Hello Have you considered the incorporation of addons created by the community within this plugin? I explain better, as well as the case of Excalidraw plugin [obsidian-excalidraw-plugin/ea-scripts at master · zsviczian/obsidian-excalidraw-plugin](https://github.com/zsviczian/obsidian-excalidraw-plugin/tree/master/ea-scripts) within it is possible to install scripts, this is what I mean by “addons inside this plugin” is only a possibility, maybe this will lighten a little the load of special features that the community requests 😉 thanks in advance for reading 👍

**Comment by ronzulu** [comment]
Thanks for the suggestion, I've only been around for a year, so not sure if this has been thought about before hand – but not since then.

I only have a chance for a quick reply, but I would think that the effort involved in creating infrastructure for a plug-in would exceed the effort of actually implementing the requested features.

I doubt that this would help in terms of bug fixing, and it would only help with special features as you say. However, it seems that the features being requested are over all  different areas of the plug-in functionality.

Supporting a plug-in architecture involves providing a robust, well documented and testable API set, and given the variety of requests this does not seem feasible.

Is there a subset of the community requests that are within a specific area of the plug-in that your idea would have the most benefit?

Cheers
Ronny

**Comment by LeonelRFF** [comment]
Thanks for answering, I was thinking of implementing this in the future and long term 💪 because if so the community could develop its own plugin and connect to "spaced-repetition" I say this for functions like:
* https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/34
* https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/80
* https://github.com/st3v3nmw/obsidian-spaced-repetition/issues/602
And others that will arise, I explain my point better with an example, assuming that I am making a drawing in excalidraw (or it can also be canvas) and I hide fragments that I want to review, that is easy, but the data of whether it was Hard, good or easy could be sent to space-repetition so that when reviewing the content is displayed according to this
and respond to this "card" It is a dynamic way of reviewing something as similar as physical paper 😊

https://github.com/user-attachments/assets/54135596-ca20-43b6-82f8-05b483e0d1d5

**Comment by ronzulu** [comment]
Is this what you want to see during a review?

![image](https://github.com/user-attachments/assets/f8770cd1-0481-4dbc-a2af-f2dc3a75437b)

And then after clicking "Show Answer"

![image](https://github.com/user-attachments/assets/08ada2a9-3b96-494b-b272-5f20445faa9e)

---

And in the following, what is the significance that there are 2 boxes called "Question 01" and 2 boxes called "Answer 01"

Is there any significance to the red dashed line?

![image](https://github.com/user-attachments/assets/22792b3b-7f28-42d2-9b43-873f6e834010)

**Comment by LeonelRFF** [comment, marked as answer]
My idea is more to review the notes as I did in my physical notebook, moving between the notes and what I had to remember, I could add the "transitions" from canvas or excalidraw to space-repetition, but if it is not possible in principle and some additional plugin is needed then I will see at another time if I can do it, in order to use the FSRS algorithm, which from what I see will be implemented soon

https://github.com/user-attachments/assets/611c6e6e-71fa-4e90-8139-52714f68fac9




The "Question 01" and 2 boxes called "Answer 01" and the red line are just for the example, it is not important,
Here is another example of what I mean, shown with Canvas, to illustrate how it would look.

https://github.com/user-attachments/assets/50076d47-78bf-48b8-91c0-724f3d209477
